Source/Core/DolphinNoGUI/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+if(APPLE)
+	set(DOLPHIN_EXE_BASE Dolphin)
+else()
+	set(DOLPHIN_EXE_BASE dolphin-emu)
+endif()
+
+set(NOGUI_SRCS MainNoGUI.cpp)
+
+list(APPEND LIBS core uicommon cpp-optparse)
+
+if(USE_X11 OR ENABLE_HEADLESS)
```

Comment:
Not sure if we want that USE_X11 conditional or the ENABLE_HEADLESS flag long term.
If the target is important, it should always be available in the workspace on all platforms where it can be built.
Could you at least try to remove that USE_X11 one or move the ENABLE_HEADLESS to the top of the file like so:
```
if(NOT ENABLE_HEADLESS)
  return()
endif()
```